app/models/manageiq/providers/redhat/inventory/parser/infra_manager.rb

```diff
@@ -0,0 +1,515 @@
+                 else
+                   logical_units = storagedomain.dig(:storage, :volume_group, :logical_units)
+                   logical_unit =  logical_units && logical_units.first
+                   logical_unit && logical_unit.id
+                 end
+
+      free        = storagedomain.try(:available).to_i
+      used        = storagedomain.try(:used).to_i
+      total       = free + used
+      committed   = storagedomain.try(:committed).to_i
+      uncommitted = total - committed
+
+      ems_ref = ManageIQ::Providers::Redhat::InfraManager.make_ems_ref(storagedomain.try(:href))
+
+      persister.datacenters.find_or_build(storagedomain.id).assign_attributes(
+        :ems_ref             => ems_ref,
+        :ems_ref_obj         => ems_ref,
+        :name                => storagedomain.try(:name),
+        :store_type          => storage_type,
+        :storage_domain_type => storagedomain.dig(:type, :downcase),
+        :total_space         => total,
+        :free_space          => free,
+        :uncommitted         => uncommitted,
+        :multiplehostaccess  => true,
+        :location            => location,
+        :master              => storagedomain.try(:master)
+      )
+    end
+  end
+
+  def datacenters
+    collector.datacenters.each do |datacenter|
+      ems_ref = ManageIQ::Providers::Redhat::InfraManager.make_ems_ref(datacenter.href)
+
+      persister.datacenters.find_or_build(datacenter.id).assign_attributes(
+        :name        => datacenter.name,
+        :type        => 'Datacenter',
+        :ems_ref     => ems_ref,
+        :ems_ref_obj => ems_ref,
+        :uid_ems     => datacenter.id,
+      )
+    end
+  end
+
+  def hosts
+    dcs = collector.datacenters
+
+    collector.hosts.each do |host|
+      host_id = host.id
+
+      power_state = host_inv.status
+      power_state, connection_state = case power_state
+                                      when 'up'             then %w(on connected)
+                                      when 'maintenance'    then [power_state, 'connected']
+                                      when 'down'           then %w(off disconnected)
+                                      when 'non_responsive' then %w(unknown connected)
+                                      else [power_state, 'disconnected']
+                                      end
+
+      hostname = host.address
+      hostname = hostname.split(',').first
+
+      nics = collector.collect_host_nics(host)
+      ipaddress = host_to_ip(nics, hostname) || host.address
+
+      ipmi_address = nil
+      if host.dig(:power_management, :type).to_s.include?('ipmi')
+        ipmi_address = host.dig(:power_management, :address)
+      end
+
+      host_os_version = host.dig(:os, :version)
+      ems_ref = ManageIQ::Providers::Redhat::InfraManager.make_ems_ref(host_inv.href)
+
+      clusters = persister.clusters.lazy_find(host.dig(:cluster, :id))
```

Comment:
if you want to access the clusters directly do:

cluster = persister.clusters.find(host.dig(:cluster, :id)) # should return just 1